src/cpp/shared/combat/combat_spells.cpp

```diff
@@ -632,4 +665,459 @@ void combatManager::CastSpell(int proto_spell, int hexIdx, int isCreatureAbility
   }
   WaitEndSample(res, res.sample);
   this->CheckChangeSelector();
+}
+
+CURSOR_DIRECTION combatManager::GetCursorDirection(int screenX, int screenY, int hex) {
```

Comment:
Was this decompiled code?

If so, what changes did you make.

If can you do some things to make this more clear?  E.g.: Use functions to make lines 674-678 more clear.

Reply:
Added a general comment on this func. I grabbed a chunk of CheckSetMouseDirection and added a switch in the end so
If you insist I'll go through the function to see how it actually manages to calculate these so-called "clock hours" and refactor it